pkg/errorfactory/errorfactory.go

```diff
@@ -19,60 +19,98 @@ import "emperror.dev/errors"
 // ResourceNotReady states that resource is not ready
 type ResourceNotReady struct{ error }
 
+func (e ResourceNotReady) Unwrap() error { return e.error }
```

Comment:
I think it would be helpful to add a short comment on top of this exported method about why we need this so it is more self-explanatory

Reply:
We'd be adding a lot of identical comments (one for each Unwrap method) for what is optional standard library-related behaviour (to be able to do what is described here: https://go.dev/blog/go1.13-errors).

I took a look at this example in the standard library: https://cs.opensource.google/go/go/+/refs/tags/go1.19.5:src/io/fs/fs.go;l=252

I'm not opposed to adding this but I'm not convinced it helps and I struggle to imagine a nice comment ... like "Unwrap() allows compliance with Go1.13+ error chaining behaviour" (many times) ?

Reply:
Well, you don't have to add the same comment in every `Unwrap()` function IMO, just adding the comment to the first one in the file would be enough and I think "Unwrap() allows compliance with Go1.13+ error chaining behavior" is a great comment to add - at least people would know what this is for without chasing back which interface we are trying to implement with adding this method

Reply:
Sounds good to me. Thanks!

Done in a99daa5a12a2e2d172c964f4e7f4b360a6ae1816.